Extensions/ArtifactEngine/Providers/filesystemProvider.ts

```diff
@@ -126,17 +128,28 @@ export class FilesystemProvider implements models.IArtifactProvider {
         return promise;
     }
 
-    private ensureDirectoryExistence(folder) {
+    private ensureDirectoryExistence(folder): boolean {
         if (!this._createdFolders.hasOwnProperty(folder)) {
-            var dirName: string = path.dirname(folder);
             if (fs.existsSync(folder)) {
-                return;
+                return true;
+            }
+
+            var dirName: string = path.dirname(folder);
+            if (dirName === folder) {
+                return false;
             }
 
-            this.ensureDirectoryExistence(dirName);
-            fs.mkdirSync(folder);
-            this._createdFolders[folder] = true;
+            if (this.ensureDirectoryExistence(dirName)) {
+                fs.mkdirSync(folder);
```

Comment:
will this throw in case of error in creating folder?

Reply:
Yes, this will throw. I will add checks to ensure it is cached.